packages/bitcore-wallet-service/src/lib/pushnotificationsservice.ts

```diff
@@ -443,19 +446,38 @@ export class PushNotificationsService {
     };
   }
 
+  _getAccessToken() {
+    const MESSAGING_SCOPE = 'https://www.googleapis.com/auth/firebase.messaging';
+    const SCOPES = [MESSAGING_SCOPE];
+
+    return new Promise(function(resolve, reject) {
+      const key = require(config.pushNotificationsOpts.fcmGoogleCredentialsPath);
+      const jwtClient = new google.auth.JWT(key.client_email, null, key.private_key, SCOPES, null);
+      jwtClient.authorize(function(err, tokens) {
+        if (err) {
+          reject(err);
+          return;
+        }
+        resolve(tokens.access_token);
+      });
+    });
+  }
+
   _makeRequest(opts, cb) {
-    this.request(
-      {
-        url: this.pushServerUrl + '/send',
-        method: 'POST',
-        json: true,
-        headers: {
-          'Content-Type': 'application/json',
-          Authorization: 'key=' + this.authorizationKey
+    this._getAccessToken().then(access_token => {
+      this.request(
+        {
+          url: this.pushServerUrl,
+          method: 'POST',
+          json: true,
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: 'Bearer ' + access_token
+          },
+          body: opts
         },
-        body: opts
-      },
-      cb
-    );
+        cb
```

Comment:
I guess we're not returning `cb` callback to the main function `_sendPushNotifications`. It's important to return any error and handle it better.

My suggestion is to add `catch` and return `cb(err)` to catch error from the request or the access token 

```
this._getAccessToken().then(access_token => {
      this.request(
        {
          url: this.pushServerUrl,
          method: 'POST',
          json: true,
          headers: {
            'Content-Type': 'application/json',
            Authorization: 'Bearer ' + access_token
          },
          body: opts
       }, cb);
    }).catch(err => {
      return cb(err);
    });
```